src/common/harvest/harvester.js

```diff
@@ -35,6 +35,14 @@ export class Harvester {
       this.initializedAggregates.forEach(aggregateInst => this.triggerHarvestFor(aggregateInst, { isFinalHarvest: true }))
       /* This callback should run in bubble phase, so that that CWV api, like "onLCP", is called before the final harvest so that emitted timings are part of last outgoing. */
     }, false)
+
+    this.agentRef.ee.on('event-buffered', (featureAgg) => {
```

Comment:
This system can be reworked to avoid ee, so do consider that approach.

- This func can be a helper method _somewhere_ (maybe this isn't the best place) that's called directly when an event is added
- The process of adding an event should pass the `evaluateHarvest` returned info to this helper
- Since `EventBuffer.add` is modified specifically, `(!this.events || this.events.StorageClass !== EventBuffer)` check wouldn't be necessary if the work done here is executed in there instead

There's a path here to lesson passing featureAgg around and making this more code than it needs to be.